idaes/core/base/extended_control_volume0d.py

```diff
@@ -0,0 +1,106 @@
+
+        Arguments are supported for compatibility with other forms but must be False
+        or None otherwise an Exception is raised.
+
+        Args:
+            has_heat_of_reaction: whether terms for heat of reaction should
+                be included in enthalpy balance
+            has_heat_transfer: whether terms for heat transfer should be
+                included in enthalpy balances
+            has_work_transfer: whether terms for work transfer should be
+                included in enthalpy balances
+            has_enthalpy_transfer: whether terms for enthalpy transfer due to
+                mass transfer should be included in enthalpy balance. This
+                should generally be the same as the has_mass_transfer
+                argument in the material balance methods
+            custom_term: a Python method which returns Pyomo expressions representing
+                custom terms to be included in enthalpy balances.
+                Method should accept time and phase list as arguments.
+
+        Returns:
+            Constraint object representing enthalpy balances
+        """
+        if has_heat_transfer:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option requires that has_heat_transfer is False. "
+                "If you are trying to solve for heat duty to achieve isothermal operation, please use "
+                "a full energy balance and add a constraint to equate inlet and outlet temperatures."
+            )
+        if has_work_transfer:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option requires that has_work_transfer is False. "
+                "If you are trying to solve for work under isothermal operation, please use "
+                "a full energy balance and add a constraint to equate inlet and outlet temperatures."
+            )
+        if has_enthalpy_transfer:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option does not support enthalpy transfer."
+            )
+        if has_heat_of_reaction:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option requires that has_heat_of_reaction is False. "
+                "If you are trying to solve for heat duty to achieve isothermal operation, please use "
+                "a full energy balance and add a constraint to equate inlet and outlet temperatures."
+            )
+        if custom_term is not None:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option does not support custom terms."
+            )
+
+        # Add isothermal constraint
+        @self.Constraint(self.flowsheet().time, doc="Energy balances")
+        def enthalpy_balances(b, t):
+            return b.properties_in[t].temperature == b.properties_out[t].temperature
```

Comment:
Can we rename this constraint to something that better describe what it does? Otherwise you might, for example, get enthalpy scaling on a constraint that should have temperature scaling.

Reply:
I debated whether the change the name - my reason for keeping it the same was for consistency of naming between different configurations (so that the "energy balance" would always have the same name no matter what option you chose).

However, I did not think about scaling, and having a descriptive name would be far more useful there for people doing custom scaling. I'll change it.